deploy-service/common/src/main/java/com/pinterest/deployservice/handler/PingHandler.java

```diff
@@ -195,7 +195,9 @@ boolean canDeploy(EnvironBean envBean, String host, AgentBean agentBean) throws
         // TODO use ecache to optimize
         // Make sure we do not exceed allowed number of concurrent active deploying agent
         String envId = envBean.getEnv_id();
-        long parallelThreshold = envBean.getMax_parallel();
+        long totalHosts = environDAO.countTotalCapacity(envBean.getEnv_id(), envBean.getEnv_name(),
```

Comment:
Does the totalHosts contain newly provisioned hosts? If yes then it may be dangerous. If suddenly a lot of new hosts are being launched, totalHosts could be bigger than # of hosts that can serve traffic and possibly allow more hosts to deploy.